src/kernel/include/component.h

```diff
@@ -41,7 +43,8 @@ comp_activate(struct captbl *t, capid_t cap, capid_t capin, capid_t captbl_cap,
 	ctc = (struct cap_captbl *)captbl_lkup(t, captbl_cap);
 	if (unlikely(!ctc || ctc->h.type != CAP_CAPTBL || ctc->lvl > 0)) return -EINVAL;
 	ptc = (struct cap_pgtbl *)captbl_lkup(t, pgtbl_cap);
-	if (unlikely(!ptc || ptc->h.type != CAP_PGTBL || ptc->lvl > 0)) return -EINVAL;
+	/* PRY : the top-level must contain metadata */
+	if (unlikely(!ptc || ptc->h.type != CAP_PGTBL || COS_PGTBL_TYPE(ptc->pgtbl.type_addr)!=COS_PGTBL_MPUMETA)) return -EINVAL;
```

Comment:
This code won't work for x86, so we need to add an abstraction here.  As I noted earlier, perhaps better as

`if (unlikely(!ptc || ptc->h.type != CAP_PGTBL || !COS_PGTBL_ROOT(ptc)))`...